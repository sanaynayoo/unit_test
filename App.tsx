import React from 'react';
import {Provider} from 'react-redux';

import {store} from './src/store/root.store';
import AppNavigator from './src/navigations/appNavigator';

const App = () => {
  return (
    <Provider store={store}>
      <AppNavigator />;
    </Provider>
  );
};

export default App;
