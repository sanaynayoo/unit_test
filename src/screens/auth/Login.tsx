import React, {useState} from 'react';
import {View, Text, StyleSheet, TouchableOpacity} from 'react-native';

const Login = () => {
  const [count, setCount] = useState(0);

  const onIncreaseHandler = () => {
    console.log('Hello Increase', count);
    setCount(pre => pre + 1);
  };

  const onDecreaseHandler = () => {
    console.log('Hello Decrease', count);
    setCount(pre => pre - 1);
  };

  return (
    <View style={styles.container}>
      <Text>Count Numbers : {count}</Text>
      <View style={styles.incContainer}>
        <TouchableOpacity
          style={styles.btnContainer}
          activeOpacity={0.8}
          onPress={onIncreaseHandler}>
          <Text style={styles.title}>Increase</Text>
        </TouchableOpacity>
        <TouchableOpacity
          style={styles.btnContainer}
          activeOpacity={0.8}
          onPress={onDecreaseHandler}>
          <Text style={styles.title}>Decrease</Text>
        </TouchableOpacity>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  incContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
  },
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    paddingTop: 20,
  },
  title: {
    fontSize: 13,
    color: '#ffffff',
  },
  btnContainer: {
    marginTop: 50,
    width: '30%',
    paddingHorizontal: 20,
    paddingVertical: 8,
    marginHorizontal: 10,
    alignItems: 'center',
    borderRadius: 100,
    backgroundColor: '#ff8800',
    shadowOffset: {width: 0, height: 8},
    shadowColor: '#000000',
    shadowOpacity: 0.1,
    shadowRadius: 10,
    elevation: 3,
  },
});

export default Login;
