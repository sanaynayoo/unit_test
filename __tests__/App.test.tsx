import 'react-native';
import React from 'react';

import App from '../App';
import About from '../src/screens/about/About';
import Login from '../src/screens/auth/Login';
import Home from '../src/screens/home/Home';

// render must be required
import {create} from 'react-test-renderer';

const appTree = create(<App />);
const AboutTree = create(<About />);
const LoginTree = create(<Login />);
const HomeTree = create(<Home />);

it('render correctly', () => {
  expect(appTree).toMatchSnapshot();
  expect(AboutTree).toMatchSnapshot();
  expect(LoginTree).toMatchSnapshot();
  expect(HomeTree).toMatchSnapshot();
});
