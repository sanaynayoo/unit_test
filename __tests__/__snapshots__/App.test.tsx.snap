// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<View>
  <Text>
    About
  </Text>
</View>
`;

exports[`render correctly 2`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "paddingTop": 20,
    }
  }
>
  <Text>
    Login
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#ff8800",
        "borderRadius": 100,
        "elevation": 3,
        "marginTop": 50,
        "opacity": 1,
        "paddingHorizontal": 20,
        "paddingVertical": 8,
        "shadowColor": "#000000",
        "shadowOffset": {
          "height": 8,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 10,
        "width": "50%",
      }
    }
  >
    <Text
      style={
        {
          "color": "green",
          "fontSize": 13,
        }
      }
    >
      Next
    </Text>
  </View>
</View>
`;

exports[`render correctly 3`] = `
<View>
  <Text>
    Home
  </Text>
</View>
`;
